support make clean option

## Feature Request

### Describe your feature request related problem

<!-- A description of what the problem is. -->

### Describe the feature you'd like

<!-- A description of what you want to happen. -->

please support `make clean` option.

currently:
```
$ git lg
* b944a8f - (grafted, HEAD -> main, origin/main, origin/HEAD) fix: remove mirrot test (#339) (5 days ago) <xhe>
$ make clean
make: *** No rule to make target `clean'.  Stop.
```

### Describe alternatives you've considered

<!-- A description of any alternative solutions or features you've considered. -->

### Teachability, Documentation, Adoption, Migration Strategy

<!-- If you can, explain some scenarios how users might use this, or situations in which it would be helpful. Any API designs, mockups, or diagrams are also helpful. -->

if compiling interrupts  (`^Cmake: *** [lint] Interrupt
`), we can use `make clean` to clean up legacy files. 
or clean up binary files, so that we can compile again without removing the current directory and git clone again.  like what we do in tiup.  https://github.com/pingcap/tiup/blob/master/Makefile#L102C1-L106C47

--- a/Makefile
+++ b/Makefile
@@ -27,7 +27,7 @@ endif
 IMAGE_TAG ?= latest
 EXECUTABLE_TARGETS := $(patsubst cmd/%,cmd_%,$(wildcard cmd/*))
 
-.PHONY: cmd_% test lint docker docker-release golangci-lint gocovmerge
+.PHONY: cmd_% test lint docker docker-release golangci-lint gocovmerge clean
 
 default: cmd
 
@@ -78,6 +78,9 @@ test: gocovmerge
 	tail -1 .cover.func
 	rm -f .cover.*
 	go tool cover -html=.cover -o .cover.html
+
+clean:
+	rm -rf bin dist grafonnet-lib
 
 docker:
 	docker build -t "$(DOCKERPREFIX)tiproxy:$(IMAGE_TAG)" --build-arg "GOPROXY=$(shell go env GOPROXY)" --build-arg "VERSION=$(VERSION)" --build-arg "COMMIT=$(COMMIT)" -f docker/Dockerfile .

